### PR TITLE
[Modules] Changed to updated API and also added new feature properties

### PR DESCRIPTION
**Pull request by prasanjeets**
# Description

>Thank you for your contribution !

> The latest API version for Microsoft.RecoveryServices/vaults and its child resources such as Microsoft.RecoveryServices/vaults/backupconfig and Microsoft.RecoveryServices/vaults/backupPolicies are outdated. at the moment, the latest API version is 2022-02-01. Same has been updated,  also updated new feature properties and tested the same.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

**Review by eriqua — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please regenerate all related readme files after changing the templates leveraging the utility in `utilities\tools\Set-ModuleReadMe.ps1`

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md [hunk 1]
```diff
@@ -36,6 +37,10 @@ This module deploys a Protection Container for a Recovery Services Vault
 | `sourceResourceId` | string | `''` |  | Resource ID of the target resource for the Protection Container. |
 
 
+### Parameter Usage: `<ParameterPlaceholder>`
+
+// TODO: Fill in Parameter usage
+
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:43]: Please either add parameter usage if needed or remove this placeholder

File: arm/Microsoft.RecoveryServices/vaults/readme.md [hunk 2]
```diff
@@ -59,38 +60,9 @@ This module deploys a recovery service vault.
 | `userAssignedIdentities` | object | `{object}` |  | The ID(s) to assign to the resource. |
 
 
-### Parameter Usage: `backupStorageConfig`
+### Parameter Usage: `<ParameterPlaceholder>`
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/readme.md:63]: A lot of parameter usage snippets got lost in the readme update. Please double check and add them back.

File: arm/Microsoft.RecoveryServices/vaults/readme.md [hunk 3]
```diff
@@ -1,30 +1,31 @@
-# Recovery Services Vaults `[Microsoft.RecoveryServices/vaults]`
+# RecoveryServices Vaults `[Microsoft.RecoveryServices/vaults]`
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/readme.md:1]:
```suggestion
# Recovery Services Vaults `[Microsoft.RecoveryServices/vaults]`
```

File: arm/Microsoft.RecoveryServices/vaults/readme.md [hunk 4]
```diff
@@ -1,30 +1,31 @@
...
 
-This module deploys a recovery service vault.
+This module deploys RecoveryServices Vaults.
+// TODO: Replace Resource and fill in description
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/readme.md:4]:
```suggestion
This module deploys a recovery services vault.
```

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md [hunk 5]
```diff
@@ -1,18 +1,19 @@
-# RecoveryServicesProtectionContainer `[Microsoft.RecoveryServices/vaults/protectionContainers]`
+# RecoveryServices Vaults ProtectionContainers `[Microsoft.RecoveryServices/vaults/protectionContainers]`
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:1]:
```suggestion
# Recovery Services Vaults Protection Containers `[Microsoft.RecoveryServices/vaults/protectionContainers]`
```

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
 
-This module deploys a Protection Container for a Recovery Services Vault
+This module deploys RecoveryServices Vaults ProtectionContainers.
+// TODO: Replace Resource and fill in description
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:4]:
```suggestion
This module deploys a recovery services vault protection container.
```

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md (hunk 1 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:46]: Done

File: arm/Microsoft.RecoveryServices/vaults/readme.md (hunk 2 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/readme.md]: Done

File: arm/Microsoft.RecoveryServices/vaults/readme.md (hunk 3 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/readme.md]: Done

File: arm/Microsoft.RecoveryServices/vaults/readme.md (hunk 4 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/readme.md]: Done

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md (hunk 5 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:1]: Done

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md (hunk 6 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:4]: Done

File: arm/Microsoft.RecoveryServices/vaults/deploy.bicep [hunk 7]
```diff
@@ -239,6 +239,7 @@ module rsv_backupConfig 'backupConfig/deploy.bicep' = if (!empty(backupConfig))
     storageTypeState: contains(backupConfig, 'storageTypeState') ? backupConfig.storageTypeState : 'Locked'
     isSoftDeleteFeatureStateEditable: contains(backupConfig, 'isSoftDeleteFeatureStateEditable') ? backupConfig.isSoftDeleteFeatureStateEditable : true
     enableDefaultTelemetry: enableChildTelemetry
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/deploy.bicep:241]: Please remove this line as it has been correctly replaced below
```suggestion
```

File: arm/Microsoft.RecoveryServices/vaults/deploy.bicep [hunk 8]
```diff
@@ -195,7 +195,9 @@ module rsv_backupStorageConfiguration 'backupStorageConfig/deploy.bicep' = if (!
     recoveryVaultName: rsv.name
     storageModelType: backupStorageConfig.storageModelType
     crossRegionRestoreFlag: backupStorageConfig.crossRegionRestoreFlag
-    enableDefaultTelemetry: enableReferencedModulesTelemetry
+    enableDefaultTelemetry: enableChildTelemetry
```
Comment (eriqua) [inline arm/Microsoft.RecoveryServices/vaults/deploy.bicep:198]: Please replace any occurrence of `enableChildTelemetry` with `enableReferencedModulesTelemetry`

Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/deploy.bicep]: Done

File: arm/Microsoft.RecoveryServices/vaults/deploy.bicep (hunk 7 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/deploy.bicep]: Done

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md (hunk 1 above)
Comment (eriqua) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:43]: Could you kindly double check? I still see the old code

Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:43]: Not sure, I can see the correct code in my branch

File: arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md (hunk 5 above)
Comment (prasanjeets) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:1]: Not sure, I can see the correct code in my branch

Comment (MrMCake) [reply arm/Microsoft.RecoveryServices/vaults/protectionContainers/readme.md:1]: See other comment

**Pull request closed without merging**